psycopg_pool/psycopg_pool/pool_async.py

```diff
@@ -381,10 +389,13 @@ async def check(self) -> None:
             else:
                 await self._add_to_pool(conn)
 
-    def reconnect_failed(self) -> None:
+    async def reconnect_failed(self) -> None:
         """
         Called when reconnection failed for longer than `reconnect_timeout`.
         """
+        if asyncio.iscoroutinefunction(self._reconnect_failed):
+            await self._reconnect_failed(self)
+            return
```

Comment:
I'd rather the sync option to be the `else` branch of this `if`.

Reply:
Ah, sure, can do.